### DISCUSSION TITLE
Erro ao instalar biblioteca

### DISCUSSION THREAD

**Discussion by sandrosato** [BIPES/BIPES#116] [Português] [answered]
Antes estava funcionando, mas agora as bibliotecas não estão sendo instaladas.

This will automatic download and install the library on the connected board: mcp23017. 
Internet is required for this operation. Install results will be shown on console tab: webserial

O arquivo na aba files fica em branco.

>>> import struct
>>> f=open('mcp23017.py', 'w')
>>> f.write('# Copyright (c) 2014 Adafruit Industries
Traceback (most recent call last):
  File "<stdin>", line 1
SyntaxError: invalid syntax

**Comment by gastmaier** [comment, marked as answer]
Oi tudo bem?
Reverti a modificação que causou o problema no beta2, quando for testar, certifique de recarregar completamente a página (CTRL+F5).

Além disso, queria dizer que estamos trabalhando em uma versão melhorada na branch "third". Já tem um alpha em [bipes.net.br/3](https://bipes.net.br/3/ide-pt-br)

(porém estou sem opção de abrir porta para WebSocket no demo no momento, necessária para que o novo easyMQTT funcione; então até  eu incluir a opção de MQTT via HTTP no código-fonte, não estará disponível no demo :/ )

**Reply by Liviems** [reply]
oiee, tudo bem? eu to tendo o mesmo problema do @sandrosato, até tentei fazer como você disse mas não deu resultado... infelizmente não consigo instalar as bibliotecas e precisava disso com urgência, por acaso você teria mais alguma dica pra me salvar? haha

**Reply by gastmaier** [reply]
> oiee, tudo bem? eu to tendo o mesmo problema do @sandrosato, até tentei fazer como você disse mas não deu resultado... infelizmente não consigo instalar as bibliotecas e precisava disso com urgência, por acaso você teria mais alguma dica pra me salvar? haha

Olá,

Você poderia fornecer o resultado que aparece no console também, para eu ver o erro?

Algumas opções para contornar o problema:

Você pode copiar o conteúdo da biblioteca que você precisa de [bipes.net.br/beta2/ui/pylibs](https://bipes.net.br/beta2/ui/pylibs) e colar no editor de arquivos do beta2 e salvar...

Ou você pode tentar pela nova versão em [bipes.net.br/3](https://bipes.net.br/3/ide-pt-br). Antes de clicar o botao para instalar a biblioteca (e depois de conectar) vá na página arquivos e com o botão direito em "Explorar dispositivo" crie uma pasta chamada "lib" (se ainda não existir), porque é nela que a nova versão instalará a biblioteca.
Então, tente instalar pelo botão.
Depois disso, pode voltar para versão beta2

**Reply by Liviems** [reply]
![image](https://user-images.githubusercontent.com/103293581/162769638-a8898303-c0de-44c8-a90f-749db3071aa4.png)
é esse erro que aparece
